sbi/inference/snl/snl.py

```diff
@@ -95,32 +75,11 @@ def __init__(
             get_potential_function=PotentialFunctionProvider(),
         )
 
-        # switch to training mode
-        self._neural_posterior.neural_net.train()
-
-        # Need somewhere to store (parameter, observation) pairs from each round.
-        self._parameter_bank, self._observation_bank = [], []
+        # XXX why not density_estimator.train(True)???
+        self._neural_posterior.neural_net.train(True)
```

Comment:
`density_estimator` is not used hereafter. So setting anything in `density_estimator` has no effect.